src/proposal.mligo

```diff
@@ -391,22 +399,22 @@ let flush(n, config, store : nat * config * storage): return =
 
 // Removes an accepted and finished proposal by key.
 let drop_proposal (proposal_key, config, store : proposal_key * config * storage): return =
-  let store = authorize_admin store in
-
   let proposal = check_if_proposal_exist (proposal_key, store) in
-  if is_voting_period_over(proposal, config.voting_period, store)
+
+  if   (sender = proposal.proposer)
+    || (sender = store.guardian && sender <> source) // Guardian cannot be equal to SOURCE
+    || is_proposal_expired_time_reached (proposal, config.proposal_expired_time)
```

Comment:
I suggest calculating the expired condition beforehand as it's needed later too (see next comment):
```suggestion
  let proposal_is_expired = is_proposal_expired_time_reached (proposal, config.proposal_expired_time) in

  if   (sender = proposal.proposer)
    || (sender = store.guardian && sender <> source) // Guardian cannot be equal to SOURCE
    || proposal_is_expired
```